bindings/python/multibody/joint/joints-models.hpp

```diff
@@ -54,41 +54,87 @@ namespace pinocchio
     }
 
     BOOST_PYTHON_FUNCTION_OVERLOADS(addJoint_proxy_overloads,addJoint_proxy,2,3)
 
+    struct JointModelCompositeConstructorVisitor : public boost::static_visitor<boost::shared_ptr<JointModelComposite> >
+    {
+      const SE3 & m_joint_placement;
+
+      JointModelCompositeConstructorVisitor(const SE3 & joint_placement)
+      : m_joint_placement(joint_placement)
+      {}
+
+      template <typename JointModelDerived>
+      boost::shared_ptr<JointModelComposite> operator()(JointModelDerived & jmodel) const
+      {
+        return boost::shared_ptr<JointModelComposite>(new JointModelComposite(jmodel,m_joint_placement));
+      }
+    }; // struct JointModelCompositeConstructorVisitor
+
+    static boost::shared_ptr<JointModelComposite> init_proxy1(const JointModelVariant & jmodel_variant)
+    {
+      return boost::apply_visitor(JointModelCompositeConstructorVisitor(SE3::Identity()), jmodel_variant);
+    }
+
+    static boost::shared_ptr<JointModelComposite> init_proxy2(const JointModelVariant & jmodel_variant,
+                                                              const SE3 & joint_placement)
+    {
+      return boost::apply_visitor(JointModelCompositeConstructorVisitor(joint_placement), jmodel_variant);
+    }
+
     template<>
     inline bp::class_<JointModelComposite>& expose_joint_model<JointModelComposite> (bp::class_<JointModelComposite> & cl)
     {
       return cl
                .def(bp::init<const size_t> (bp::args("size"), "Init JointModelComposite with a defined size"))
+               .def("__init__",
+                    bp::make_constructor(init_proxy1,
+                                         bp::default_call_policies(),
+                                         bp::args("joint_model")
+                                        ),
+                    "Init JointModelComposite from a joint"
+                   )
+               .def("__init__",
+                    bp::make_constructor(init_proxy2,
+                                         bp::default_call_policies(),
+                                         bp::args("joint_model","joint_placement")
+                                        ),
+                    "Init JointModelComposite from a joint and a placement"
+                   )
                .add_property("joints",&JointModelComposite::joints)
-               .def("addJoint",&addJoint_proxy,addJoint_proxy_overloads(bp::args("joint_model","joint_placement"),"Add a joint to the vector of joints."))
+               .add_property("jointPlacements",&JointModelComposite::jointPlacements)
+               .add_property("njoints",&JointModelComposite::njoints)
+               .def("addJoint",
+                    &addJoint_proxy,
+                    addJoint_proxy_overloads(bp::args("joint_model","joint_placement"),
+                                             "Add a joint to the vector of joints."
+                                            )[bp::return_value_policy<bp::return_by_value>()]
```

Comment:
You should not make a return by value here. It will duplicate with a copy

Reply:
You are right, I've just forced-pushed a fix, with an additional unit test to check it behaves well.
I'm now using `return_internal_reference`

Reply:
Good!